requests/adapters.py

```diff
@@ -344,13 +348,20 @@ def request_url(self, request, proxies):
         scheme = urlparse(request.url).scheme
 
         is_proxied_http_request = (proxy and scheme != 'https')
+        is_using_https_forwarding = (
+            proxies_kwargs and proxies_kwargs.get('use_forwarding_for_https', False)
+        )
+
+        is_using_https_proxy = False
         using_socks_proxy = False
         if proxy:
             proxy_scheme = urlparse(proxy).scheme.lower()
             using_socks_proxy = proxy_scheme.startswith('socks')
+            is_using_https_proxy = proxy_scheme.startswith('https')
```

Comment:
We support a number of possible socks* schemes which is the reason for using `startswith`. I'm wondering if we want to be more explicit here by checking for exactly `https`.

Reply:
Yeah, I agree this could be more simply `proxy_scheme == 'https'` but I'm wondering if there are cases I'm unaware of where someone might use `https+<somethingelse>://proxy` or `<somethingelse>+https://proxy`.  It shouldn't be too much trouble to add support for those if necessary though and on the chance those aren't easily supportable with this work, then we won't accidentally opt them in